### ISSUE TITLE
Remove impediment from issue.

### ISSUE THREAD

**Issue by paulodamaso** [zerocracy/farm#1016]
There is no way at the moment to tell zerocracy to lift impediment from a task. It should be interesting to allow impediment removal. There is the case that an impediment is created for a task but some time later the developer discover that it is not an impediment to task anymore, but it is still an issue that must be solved. The creator of the impediment should be allowed to lift this impediment from the task, removing the `on hold` status for its issue. This ticked is related to logic discussed in https://github.com/zerocracy/farm/issues/972 .

**Comment by 0crat** [comment]
@yegor256[/z](https://www.0crat.com/u/yegor256) please, pay attention to this issue

**Comment by 0crat** [comment]
@paulodamaso[/z](https://www.0crat.com/u/paulodamaso) this project will fix the problem faster if you donate a few dollars to it; just [click here](https://www.0crat.com/contrib/C3NDPUA8L) and pay via Stripe, it's very fast, convenient and appreciated; thanks a lot!

**Comment by 0crat** [comment]
Job [#1016](https://github.com/zerocracy/farm/issues/1016) is now in scope, role is `DEV`

**Comment by 0crat** [comment]
Bug was reported, see [§29](http://www.zerocracy.com/policy.html#29): +15 point(s) just awarded to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso)

**Comment by 0crat** [comment]
The job [#1016](https://github.com/zerocracy/farm/issues/1016) assigned to @krzyk[/z](https://www.0crat.com/u/krzyk), here is [why](https://www.0crat.com/footprint/C3NDPUA8L/1611056943); the budget is 30 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§8](http://www.zerocracy.com/policy.html#8) and [§9](http://www.zerocracy.com/policy.html#9); if the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html); there will be a monetary reward for this job

**Comment by krzyk** [comment]
@paulodamaso just curious, why? Is there a point in doing so? You just finish the task earlier and the impediment is irrelevant.

**Comment by paulodamaso** [comment]
@krzyk It happened to me a couple of times: after a superficial view the issues looked dependent from something else and other issue were created, but some days after I've figured that these tickets are not related, but they do have to be solved. Take a look at #972, `Scenario 2` description and `Proposed Feature`.

Anyways, looks reasonable to me that the some impediment set to some issue could be lifted; of course we can finish the task and it'll be irrelevant, as you said, but my concern is that that dependency does not exists anymore, so it should not be enforced by zerocrat.

**Comment by krzyk** [comment]
@paulodamaso but #972 is still not accepted, so I don't see a point in doing this issue before the other. Because if #972 will be rejected we will have a code that is used by no one but we will need to maintain it and later remove it.

**Comment by paulodamaso** [comment]
@krzyk I think that the issues are related but this issue does not depends on #972. For me, it's just wrong to have an impediment set to some issue even if this impediment does not exists at all. In fact, in my opinion zerocrat should enforce impediment behavior and act exactly as you described: it should not allow ticket closure if it has impediments set on it and should check and remove impediments when impeding tasks for some ticket are finished.

Looks like for me that the impediment feature is just being used to prevent zerocrat from taking away a job, cheating on the 10 day rule. It's convenient for `DEV`, because he/she could set an impediment on anything else just to keep the task, and this impediment is never checked  by anyone.

Besides that, I see this impediment feature as a tool to set task precendence, which should be used by zerocrat when choosing which jobs should be distributed among performers. For example, we should have this impediment set by `ARC` when the job is put into scope, but zerocrat would assign it to someone only when its impediments were removed. Say we have five issues, the fifth depending on all other four, but they are part of the same "work package"; in this case `ARC` could set the impediment on issue creation and zerocrat would assign it only when all other four are closed or the impediment is removed, meaning that it is not valid anymore.

Does it make sense?

**Comment by 0crat** [comment]
@krzyk[/z](https://www.0crat.com/u/krzyk) this job was assigned to you 5days ago. It will be taken away from you soon, unless you close it, see [§8](http://www.zerocracy.com/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

**Comment by krzyk** [comment]
@0crat refuse

**Comment by 0crat** [comment]
> @0crat refuse ([here](https://github.com/zerocracy/farm/issues/1016#issuecomment-397608795))

@krzyk The user @krzyk[/z](https://www.0crat.com/u/krzyk) resigned from [#1016](https://github.com/zerocracy/farm/issues/1016), please stop working. Reason for job resignation: Order was cancelled

**Comment by 0crat** [comment]
Tasks refusal is discouraged, see [§6](http://www.zerocracy.com/policy.html#6): -15 point(s) just awarded to @krzyk[/z](https://www.0crat.com/u/krzyk)

**Comment by 0crat** [comment]
@amihaiemil[/z](https://www.0crat.com/u/amihaiemil) this job was assigned to you 5days ago. It will be taken away from you soon, unless you close it, see [§8](http://www.zerocracy.com/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

**Comment by 0pdd** [comment]
@paulodamaso the puzzle [#1380](https://github.com/zerocracy/farm/issues/1380) is still not solved.

**Comment by amihaiemil** [comment]
@paulodamaso PR #1378 has been merged, can we close this? :D

**Comment by amihaiemil** [comment]
@paulodamaso ping, can we close this?

**Comment by paulodamaso** [comment]
@amihaiemil Sorry for the delay. I'd just ask you to put a puzzle in code about setting / removing multiple impediments to a job in the future, I think that would be a great feature.

**Comment by amihaiemil** [comment]
@paulodamaso please open another issue about that and discuss with the ARC there, since it is a completely new feature. Besides, I have already worked the allocated time here.

**Comment by amihaiemil** [comment]
@paulodamaso ping

**Comment by paulodamaso** [comment]
@amihaiemil Sorry, I didn't had much time to take care of this. Let's close this one for now, I'll submit a new issue later.

**Comment by 0crat** [comment]
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job completed by @amihaiemil[/z](https://www.0crat.com/u/amihaiemil), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#1016](https://github.com/zerocracy/farm/issues/1016) is now out of scope

**Comment by ypshenychka** [comment]
@0crat quality good

**Comment by 0crat** [comment]
Order was finished, quality is "good": +46 point(s) just awarded to @amihaiemil[/z](https://www.0crat.com/u/amihaiemil)

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)

**Comment by 0pdd** [comment]
@paulodamaso the puzzle [#1461](https://github.com/zerocracy/farm/issues/1461) is still not solved; solved: [#1380](https://github.com/zerocracy/farm/issues/1380).
